python/cairo-core/src/cairo_core/hash/blake2s.cairo

```diff
@@ -0,0 +1,321 @@
+
+    %{
+        B = 32
+        MASK = 2 ** 32 - 1
+        segments.write_arg(ids.data, [(ids.high >> (B * (3 - i))) & MASK for i in range(4)])
+        segments.write_arg(ids.data + 4, [(ids.low >> (B * (3 - i))) & MASK for i in range(4)])
+    %}
+
+    assert data[0] * 2 ** 96 + data[1] * 2 ** 64 + data[2] * 2 ** 32 + data[3] = high;
+    assert data[4] * 2 ** 96 + data[5] * 2 ** 64 + data[6] * 2 ** 32 + data[7] = low;
+    let data = data + 8;
+    return ();
+}
+
+// Serializes a field element in a blake2s compatible way.
+func blake2s_add_felt{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, data: felt*}(
+    num: felt, bigend: felt
+) {
+    let (high, low) = split_felt(num);
+    if (bigend != 0) {
+        blake2s_add_uint256_bigend(Uint256(low=low, high=high));
+        return ();
+    } else {
+        blake2s_add_uint256(Uint256(low=low, high=high));
+        return ();
+    }
+}
+
+// Serializes multiple field elements in a blake2s compatible way.
+// Note: This function does not serialize the number of elements. If desired, this is the caller's
+// responsibility.
+func blake2s_add_felts{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, data: felt*}(
+    n_elements: felt, elements: felt*, bigend: felt
+) -> () {
+    if (n_elements == 0) {
+        return ();
+    }
+    blake2s_add_felt(num=elements[0], bigend=bigend);
+    return blake2s_add_felts(n_elements=n_elements - 1, elements=&elements[1], bigend=bigend);
+}
+
+// Computes the blake2s hash for multiple field elements.
+func blake2s_felts{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, blake2s_ptr: felt*}(
+    n_elements: felt, elements: felt*, bigend: felt
+) -> (res: Uint256) {
+    alloc_locals;
+    let (data) = alloc();
+    let data_start = data;
+    with data {
+        blake2s_add_felts(n_elements=n_elements, elements=elements, bigend=bigend);
+    }
+    let (res) = blake2s(data=data_start, n_bytes=n_elements * 32);
+    return (res=res);
+}
+
+// Taken from https://github.com/starkware-libs/stwo-cairo/blob/main/stwo_cairo_prover/test_data/test_prove_verify_all_opcode_components/all_opcode_components.cairo
+// Forces the runner to execute the Blake2s or Blake2sLastBlock opcode with the given operands.
+// op0 is a pointer to an array of 8 felts as u32 integers of the state.
+// op1 is a pointer to an array of 16 felts as u32 integers of the message.
+// dst is a felt representing a u32 of the counter.
+// ap contains a pointer to an array of 8 felts as u32 integers of the output state.
+// Those values are stored within addresses fp-5, fp-4 and fp-3 respectively.
+// An instruction encoding is built from offsets -5, -4, -3 and flags which are all 0 except for
+// those denoting uses of fp as the base for operand addresses and flag_opcode_blake (16th flag).
+// The instruction is then written to [pc] and the runner is forced to execute Blake2s.
+func run_blake2s_opcode(is_last_block: felt, dst: felt, op0: felt*, op1: felt*) -> felt* {
```

Comment:
I don't think that we need to copy this all test function. We can just use the compiled `dw` directly